### PR TITLE
Add SSL4EO-L Benchmark to docs

### PR DESCRIPTION
**Pull request by nilsleh**
I think we forgot to add the SSL4EO Benchmark Dataset/Datamodule to the Docs. Do you want to put entries for all possible constillations in the csv file?

### CODE REVIEW

**Comment by adamjstewart** [comment]
Good catch! Should we also add this to `non_geo_datasets.csv`?

Yes, we should have an entry for every family of datasets in both the docs and the tables.

File: docs/api/non_geo_datasets.csv [hunk 1]
```diff
@@ -37,6 +37,8 @@ Dataset,Task,Source,# Samples,# Classes,Size (px),Resolution (m),Bands
 `SpaceNet`_,I,WorldView-2/3 Planet Lab Dove,"1,889--28,728",2,102--900,0.5--4,MSI
 `SSL4EO`_-L,T,Landsat,1M,-,264x264,30,MSI
 `SSL4EO`_-S12,T,Sentinel-1/2,1M,-,264x264,10,"SAR, MSI"
+`SSL4EOLBenchmark`_-TM-TOA & CDL, S, Lansat, 25k,18,264x264,30,MSI
```
Comment (nilsleh) [inline docs/api/non_geo_datasets.csv:40]: @adamjstewart If the row entry looks okay to you, I will add the other benchmark dataset combinations

Comment (adamjstewart) [reply docs/api/non_geo_datasets.csv:40]: Let's just add a single row for "Landsat & NLCD & CDL"

File: docs/api/datamodules.rst [hunk 2]
```diff
@@ -131,6 +131,11 @@ SSL4EO
 .. autoclass:: SSL4EOLDataModule
 .. autoclass:: SSL4EOS12DataModule
 
+SSL4EOLBenchmarkDataModule
```
Comment (adamjstewart) [inline docs/api/datamodules.rst:134]: Should be human-readable name, not class name. Like "SSL4EO-L Benchmark". Same in `datasets.rst` and in `non_geo_datasets.csv`

File: docs/api/non_geo_datasets.csv (hunk 1 above)
Comment (nilsleh) [reply docs/api/non_geo_datasets.csv:40]: maybe one for NLCD and one for CLD because they have different number of classes?

Comment (adamjstewart) [reply docs/api/non_geo_datasets.csv]: Sure

File: docs/api/non_geo_datasets.csv [hunk 3]
```diff
@@ -37,6 +37,8 @@ Dataset,Task,Source,# Samples,# Classes,Size (px),Resolution (m),Bands
 `SpaceNet`_,I,WorldView-2/3 Planet Lab Dove,"1,889--28,728",2,102--900,0.5--4,MSI
 `SSL4EO`_-L,T,Landsat,1M,-,264x264,30,MSI
 `SSL4EO`_-S12,T,Sentinel-1/2,1M,-,264x264,10,"SAR, MSI"
+`SSL4EO-L Benchmark`_, S, Lansat & CDL, 25k,18,264x264,30,MSI
+`SSL4EO-L Benchmark`_, S, Lansat & NLCD, 25k,14,264x264,30,MSI
```
Comment (adamjstewart) [inline docs/api/non_geo_datasets.csv:41]: NLCD has 17 classes and CDL has 134 classes (although we only evaluated on a subset in our paper). Also, I would remove the spaces around commas and capitalize the K.

Comment (nilsleh) [reply docs/api/non_geo_datasets.csv:41]: the config files that we used say 14 classes for nlcd and 18 for cdl

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**